src/companion.jl

```diff
@@ -1,11 +1,12 @@
 export Companion
+using LinearAlgebra: dot
 
 """
 [`Companion` matrix](http://en.wikipedia.org/wiki/Companion_matrix)
 
 ```julia
```

Comment:
```suggestion
```jldoctest
```

Reply:
This entire package needs `jldoctest`s and I was going to do that with a separate PR after getting to 100% code coverage and 3-arg `mul!` etc.  I'll commit your suggestion but let's leave the rest for another PR...